bitcoin/src/util/from_hex_str.rs

```diff
@@ -0,0 +1,44 @@
+
+//! FromHexStr
+//! Functionality to allow types to take hex strings as input
+
+#[cfg(feature = "no-std")]
+use crate::prelude::String;
+
+#[cfg(feature = "no-std")]
+use crate::alloc::string::ToString;
+
+
+/// Hex Parsing error
+#[derive(Debug, Eq, PartialEq, Clone)]
+#[non_exhaustive]
```

Comment:
I suspect we may not need `non_exhaustive` here but let's keep it for now.

Reply:
:+1: